opensearch-project/OpenSearch-Dashboards
# [Discover] A bunch of navigation fixes

### Description


 Noteable fixes here are:
- Replaces history api with application service's navigate method to correctly preserve state during redirection. The changes were made in:
  -  Breadcrumbs
  - Opening a saved search
  - Creating a new saved search
- Adds a hydration action to load a url state into redux if a change to url should update the redux store. This fixes the issue where browser navigation did not update redux state.

> Note: Recreated from #5123 since targeting that branch to main caused some issues during rebase.

### Issues Resolved




closes #5056
closes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5157
closes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5066
closes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5136

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/20453492/f59ac5fc-412d-4164-a12e-610974ee1735





## Testing the changes



Opening a saved search
- Change timerange
- Open a saved search
- Timerange should persist

Creating a new search
- Open a saved search
- Change timerange
- Click "New" in the menu bar
- Timerange should persist

Reset
- Open a saved search
- Change timerange
- Click "Reset" above the histogram
- Timerange should persist

Navigation
- Go to Discover
- Open a saved search
- Go back in the browser
- You should be able to go back to the page where the saved search was not open

### Check List
